### PR TITLE
Use a replica set as a config server if the server version is at least 3.1.2

### PR DESCRIPTION
**Pull request by llvtt**
Addresses #191

### CODE REVIEW

**Comment by behackett** [comment]
Huh, I guess this was never merged. We definitely need it now:

https://jira.mongodb.org/browse/SERVER-22318
https://jira.mongodb.org/browse/SERVER-22996

**Comment by llvtt** [comment]
I don't remember what was holding this up. Can someone review please?

**Comment by behackett** [comment]
I don't remember either. Working on it. :-)

File: mongo_orchestration/sharded_clusters.py [hunk 1]
```diff
@@ -56,8 +56,24 @@ def __init__(self, params):
...
             self.kwargs.update(DEFAULT_SSL_OPTIONS)
 
+        # Determine what to do with config servers via mongos version.
+        mongos = Server(name='mongos', procParams={})
+        mongos_version = mongos.version
+        mongos.cleanup()
         configsvr_configs = params.get('configsvrs', [{}])
-        self.__init_configsvr(configsvr_configs)
+        self.uses_rs_configdb = (mongos_version >= (3, 1, 2) and
+                                 len(configsvr_configs) == 1)
```
Comment (behackett) [inline mongo_orchestration/sharded_clusters.py]: What if we want more than one member in the config server replica set?

Comment (llvtt) [inline mongo_orchestration/sharded_clusters.py]: You can. `configsvrs` either contains multiple server parameter definitions or a single replica set definition (which may contain multiple entries in its `members`).

File: mongo_orchestration/sharded_clusters.py [hunk 2]
```diff
@@ -56,8 +56,24 @@ def __init__(self, params):
...
+        self.configdb_singleton = (
+            ReplicaSets() if self.uses_rs_configdb else Servers())
+        if self.uses_rs_configdb:
+            self.__init_configrs(configsvr_configs[0])
+        elif mongos_version >= (3, 1, 2) and len(configsvr_configs) != 3:
```
Comment (behackett) [inline mongo_orchestration/sharded_clusters.py]: Also need to return a useful error message when the server version is >= 3.3.2 and 3 old style config servers were specified.

File: mongo_orchestration/sharded_clusters.py (hunk 1 above)
Comment (behackett) [inline mongo_orchestration/sharded_clusters.py]: I see. Thanks.

**Comment by llvtt** [comment]
+ useful error message

**Comment by behackett** [comment]
LGTM

**Pull request merged**